CDCgov/prime-reportstream
# fix: time parser accepts single digit minutes [experience]

Currently if data is submitted to report stream at a time with single digit minutes (meaning between 0 and 9 past the hour), the submissions detail screen will report `parsing error` for the time and date of the submission. This is due to a bug with the date parsing on the front end which is fixed by this change.

Note, a test was deleted here. The test was designed to ensure that the date time parser would error out when receiving and unparseable date string representing a number. Upon fixing this bug, the test broke, as it turns out the test was relying on the bug for the expected error. As date parsing behavior varies by platform, the test case here (`Date.parse("-1")`) actually returns a valid Date object in the jest implementation. Test was removed as, again, it's platform specific so doesn't really test anything meaningful, and as the functionality under test is adequately handled by the previous test.

Unticketed

## Test Steps:

1. start servers o with `yarn run start:localdev` from `frontend-react` and `./gradlew run` from `prime-router`
2. visit http://localhost:3000/admin/settings and log in as necessary
4. activate the `user-upload` feature flag via the `admin -> feature flags` navigation
5. refresh
3. set up your user up to spoof 'hcintegrations'
4. wait until 0-10 past the hour.
8. click `User Upload` in the nav bar under `Upload & Validate`
9. upload a valid hl7 file
10. once upload is complete and successful, click submissions
11. your upload should appear at the top of the list, click it
12. _VERIFY_: detail page shows `Parsing Error` under `Transmission Date` and `Transmission Time`
13. repeat steps 1 - 5 on this branch
14. visit submissions, and click into the same detail as before
15. _VERIFY_: you see valid date and time date under `Transmission Date` and `Transmission Time`

## Screenshots

Before:
<img width="1370" alt="Screen Shot 2022-07-27 at 10 50 02 AM" src="https://user-images.githubusercontent.com/92806979/181279003-a6b0fcfa-b30b-441e-a663-723471e56d17.png">

After:

<img width="1372" alt="Screen Shot 2022-07-27 at 10 52 31 AM" src="https://user-images.githubusercontent.com/92806979/181278956-a55fda6a-5570-4c7c-9f75-4b157f3ca5cd.png">


## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing
